### PR TITLE
Adding tiramisu paper

### PR DESCRIPTION
**Pull request by czotti**
I have also a gist link with a notebook about the implementation that I can share if people want.

### CODE REVIEW

**Comment by lemairecarl** [comment]
Let me see that gist!

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+## Results
+
+Overall the model if performing as close as the other state-of-the-art model with far less parameters (100x less for some of them).
+
+### Issues
+
+Even if the model have far less parameters than the others you should keep in mind that he still have far more layers than a regular U-Net so the memory foot print is huge when you want to train it.
+
+If you run it on theano you should use some `THEANO_FLAGS` like `optimizer_includiin=fusion` to decrease the memory foot print and use a smaller batch_size.
```
Comment (Dref360) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: including*

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  venue:   "arXiv 2016"
+---
+
+This papers implement all the well known advanced in the image segmentation community and make it work together.
+
+List of the methods used:
+
+* Dense blocks
+* U-Net architecture
+* Using IOU loss
```
Comment (Dref360) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: The loss seems to be only cross_entropy : https://github.com/SimJeg/FC-DenseNet/blob/master/train.py#L101

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+## Results
+
+Overall the model if performing as close as the other state-of-the-art model with far less parameters (100x less for some of them).
+
+### Issues
+
+Even if the model have far less parameters than the others you should keep in mind that he still have far more layers than a regular U-Net so the memory foot print is huge when you want to train it.
+
+If you run it on theano you should use some `THEANO_FLAGS` like `optimizer_includiin=fusion` to decrease the memory foot print and use a smaller batch_size.
+
```
Comment (Dref360) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: Add a section for https://github.com/SimJeg/FC-DenseNet/ and your gist.

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+layout: review
+title:  "The One Hundred Layers Tiramisu: Fully Convolutional DenseNets for Semantic Segmentation"
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: Just to be coherence with the other CNN papers, please put the name of the network first :
Tiramisu-Net : The One Hundred Layers Tiramisu: Fully Convolutional DenseNets for Semantic Segmentation

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+author: Clément Zotti
+pdf:   "https://arxiv.org/pdf/1611.09326.pdf"
+
+cite:
+  authors: "S.Jégou, M.Drozdzal, D.Vasquez, A.Romero, Y.Bengio"
+  title:   "The One Hundred Layers Tiramisu: Fully Convolutional DenseNets for Semantic Segmentation"
+  venue:   "arXiv 2016"
+---
+
+This papers implement all the well known advanced in the image segmentation community and make it work together.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: This paper implements the well-known advances in deep learning segmentation methods and ...

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+List of the methods used:
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: The proposed network implements the following things:

File: deep-learning/_posts/2017-05-17-tiramisu.md (hunk 5 above)
Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: I suggest a rewording of this sentence:

This papers implementS all the advanceS well known in the image segmentation community and make **them** work together.

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+* Dense blocks
+* U-Net architecture
+* Using IOU loss
+
+
+## Dense blocks
+
+The dense block provide a low increase of parameters when keeping a good feature extraction.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: provides
when keeping a good feature extraction? What does that means?

Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: provides
I suggest: "while keeping strong feature extraction"

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+<div align="middle">
+     <img src="/deep-learning/images/tiramisu/dense_blocks.png">
+</div>
+
+## U-Net
+
+The U-Net is well known to have a good performance on various images segmentation dataset.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: datasets

Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: imagE segmentation datasetS

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+## Architecture
+
+The network architecture is a straight U-Net with more stage of comvolution in each steps.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: more convolution stages

btw, what is a stage?  do you mean a layer?

Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: more stageS of coNvolution in each steP

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+<div align="middle">
+     <img src="/deep-learning/images/tiramisu/architecture.png">
+</div>
+
+In table, $$m$$ represent the number of feature maps output, DB mean dense block, TD mean transition down and TU mean transition up.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: In the following table
represents
means

Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: In this table, m representS

Replace all "mean" by "means"

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+Here is the internal description of all the blocks defined above.
+
+<div align="middle">
+     <img src="/deep-learning/images/tiramisu/layers.png">
+</div>
+
+
+## Results
+
+Overall the model if performing as close as the other state-of-the-art model with far less parameters (100x less for some of them).
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: IS performing close to the other state-of-the-art modelS

Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: the model is as accurate than other state-of-the-art models while having far...

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+### Issues
+
+Even if the model have far less parameters than the others you should keep in mind that he still have far more layers than a regular U-Net so the memory foot print is huge when you want to train it.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: has fewer parameters than the other models,

Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: it still has
footprint

File: deep-learning/_posts/2017-05-17-tiramisu.md (hunk 1 above)
Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: `includiin` is this correct? looks like a typo

footprint

File: deep-learning/_posts/2017-05-17-tiramisu.md (hunk 7 above)
Comment (lemairecarl) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: I think it should be "while keeping"

File: deep-learning/_posts/2017-05-17-tiramisu.md (hunk 2 above)
Comment (czotti) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: Your right it's not the loss but one of the evaluation metrics. I correct it.

File: deep-learning/_posts/2017-05-17-tiramisu.md (hunk 7 above)
Comment (czotti) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: It mean that you have less feature maps than other big network and still extract meaningfull features by leveraging the skip connections in the dense block.

File: deep-learning/_posts/2017-05-17-tiramisu.md (hunk 9 above)
Comment (czotti) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: A stage it's the number of `layers` inside a dense block, but it's not regular convolutional layers it's the `Layer` show in the figure in the `Layers` part of the post. I can change the name which one did you prefer ? They call it `n_layers_per_block` in their code.

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+author: Clément Zotti
+pdf:   "https://arxiv.org/pdf/1611.09326.pdf"
+
+cite:
+  authors: "S.Jégou, M.Drozdzal, D.Vasquez, A.Romero, Y.Bengio"
+  title:   "The One Hundred Layers Tiramisu: Fully Convolutional DenseNets for Semantic Segmentation"
+  venue:   "arXiv 2016"
+---
+
+This papers implement all the well-known advances in deep learning segmentations methods and make it work together on these datasets [CamVid](http://mi.eng.cam.ac.uk/research/projects/VideoRec/CamVid/){:target="_blank"} and [Gatech](http://www.cc.gatech.edu/cpl/projects/videogeometriccontext/){:target="_blank"}.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: This paper implements
segmentation methods
on the CamVid and Gatech datasets.

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+Here is the internal description of all the blocks defined above.
+
+<div align="middle">
+     <img src="/deep-learning/images/tiramisu/layers.png">
+</div>
+
+
+## Results
+
+Overall the model is as accurate than the other state-of-the-art models with having far less parameters (100x less for some of them).
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: while having much fewer

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+
+### Issues
+
+Even if the model has fewer parameters than the others models, you should keep in mind that it still has far more layers than a regular U-Net so the memory footprint is huge when you want to train it.
+
+If you run it on theano you should use some `THEANO_FLAGS` like `optimizer_including=fusion` to decrease the memory footprint and use a smaller batch_size.
+
+#### Implementations
+
+This is the `Theano/Lasagne` implementations of the paper authors [FC-DenseNet](https://github.com/SimJeg/FC-DenseNet/){:target="_blank"}.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: The original Theano/Lasagne` implementations of the paper can be found here ....

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+
+And this is my notebook contribution with `Keras` [implementation](https://gist.github.com/czotti/b1e34c23a92e64490be83f3b8908bdbe){:target="_blank"}.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: This is my personal `Keras`/notebook contribution...

File: deep-learning/_posts/2017-05-17-tiramisu.md [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+author: Clément Zotti
+pdf:   "https://arxiv.org/pdf/1611.09326.pdf"
+
+cite:
+  authors: "S.Jégou, M.Drozdzal, D.Vasquez, A.Romero, Y.Bengio"
+  title:   "The One Hundred Layers Tiramisu: Fully Convolutional DenseNets for Semantic Segmentation"
+  venue:   "arXiv 2016"
+---
+
+This papers implements all the well-known advances in deep learning segmentation methods and make it work together on the [CamVid](http://mi.eng.cam.ac.uk/research/projects/VideoRec/CamVid/){:target="_blank"} and [Gatech](http://www.cc.gatech.edu/cpl/projects/videogeometriccontext/){:target="_blank"} datasets.
```
Comment (jodoin) [inline deep-learning/_posts/2017-05-17-tiramisu.md]: This paper implements
